server/lyft/decorators/audit_project_commands_wrapper.go

```diff
@@ -0,0 +1,140 @@
+package decorators
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/events"
+	"github.com/runatlantis/atlantis/server/events/models"
+	"github.com/runatlantis/atlantis/server/lyft/aws/sns"
+)
+
+// AuditProjectCommandWrapper is a decorator that notifies sns topic
+// about the state of the command. It is used for auditing purposes
+type AuditProjectCommandWrapper struct {
+	SnsWriter sns.Writer
+	events.ProjectCommandRunner
+}
+
+func (p *AuditProjectCommandWrapper) Apply(ctx models.ProjectCommandContext) models.ProjectResult {
+	id := uuid.New()
+	startTime := time.Now()
+
+	applyEvent := &ApplyEvent{
+		Version:        1,
+		ID:             id.String(),
+		RootName:       ctx.ProjectName,
+		JobType:        models.ApplyCommand,
+		Respository:    ctx.BaseRepo.FullName,
+		Environment:    ctx.Tags["environment"],
+		PullNumber:     ctx.Pull.Num,
```

Comment:
Should we keep `Repository` and `Pull` as structs so that we can add related fields within them?

For example, we'll want revision, and approved info which fit nicely in those blocks respectively.

Reply:
We could, I was modeling the schema based on IDL code, where everything is denormalizer. Do we need to update IDL as well to match the new schema?

Reply:
nah it's probably fine. Probably good to keep it consistent.